pkg/cache/scheduler/fair_sharing.go

```diff
@@ -59,30 +124,21 @@ func dominantResourceShare(node dominantResourceShareNode, wlReq resources.Flavo
 		}
 	}
 	if len(borrowing) == 0 {
-		return 0, ""
+		return drs
 	}
 
-	var drs int64 = -1
-	var dRes corev1.ResourceName
-
 	lendable := calculateLendable(node.parentHRN())
 	for rName, b := range borrowing {
 		if lr := lendable[rName]; lr > 0 {
-			ratio := b * 1000 / lr
+			ratio := float64(b) * 1000.0 / float64(lr)
```

Comment:
Also happy with float64. But want to point out its not the multiplication idea suggested by @PBundyra here: https://github.com/kubernetes-sigs/kueue/issues/6774#issuecomment-3291706679

Reply:
This is still an option, but I don't see a benefit, unless we think that float precision will be a problem, which I don't. If we switch to this implementation, we will need to use [math/big](https://pkg.go.dev/math/big) to avoid any overflow bugs

Reply:
So, let's see:
1. float64 with division is enough for all practical cases. If we ever lack precision then the consequence is known - infinite preemption loop.
2. int64  with multiply is enough for all practical cases, but technically `weight * borrowing * lendable`  could exceed MaxInt64. The consequences of overflow are hard to predict, so we would mitigate by some check 
3. math/big with multiply is fully correct, but may have some performance penalty.

I'm ok either with (1.) or (3.).